bindings/pyroot_experimental/pythonizations/src/FacadeHelpers.cxx

```diff
@@ -0,0 +1,46 @@
+// Author: Enric Tejedor CERN  04/2020
+
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+// Bindings
+#include "FacadeHelpers.hxx"
+
+// Cppyy
+#include "LowLevelViews.h"
+
+// ROOT
+#include "RtypesCore.h"
+#include "ROOT/RConfig.hxx"
+
+//////////////////////////////////////////////////////////////////////////
+/// \brief Get a buffer starting at a given address
+/// \param[in] self Always null, since this is a module function.
+/// \param[in] addr Address to create buffer from
+///
+/// Returns a cppyy LowLevelView object on the received address, i.e. an
+/// indexable buffer starting at that address.
+PyObject *PyROOT::CreateBufferFromAddress(PyObject * /* self */, PyObject *addr)
+{
+   if (!addr) {
+      PyErr_SetString(PyExc_RuntimeError, "Unable to create buffer from invalid address");
+      return NULL;
+   }
+
+   Long64_t cAddr = PyLong_AsLongLong(addr);
+   if (cAddr == -1 && PyErr_Occurred()) {
```

Comment:
A `Long64_t` can also be `-1` without being wrong. I never understood this logic from `PyLong_AsLongLong`, you do?

Reply:
Or is the `PyErr_Occured` the actual test? Then we don't need to check the value of `-1`.

Reply:
Yeah this point is actually mentioned in the docs:

https://docs.python.org/3/c-api/long.html#c.PyLong_AsLongLong

`Returns -1 on error. Use PyErr_Occurred() to disambiguate.`

So my rational here was: I do a quick check first for `-1` as return value, and only if that is true I check if `PyErr_Occurred`.

Reply:
On most of the cases, I will just check for the -1.

Reply:
Right, I always forget that the `&&` statements are executed in order -> https://stackoverflow.com/questions/5211961/how-does-c-handle-short-circuit-evaluation

Reply:
Most likely your impl is the most efficient one.